ui/StatusQ/src/statussyntaxhighlighter.cpp

```diff
@@ -47,14 +48,57 @@ void StatusSyntaxHighlighter::componentComplete()
     rule.format = codeFormat;
     highlightingRules.append(rule);
     //CODEBLOCK
+
+    //HYPERLINKS
+    //QRegularExpression to match any hyperlink in m_hyperlinks
+    hyperlinkFormat.setForeground(m_hyperlinkColor);
+    rule.pattern = hyperlinksRegularExpression();
+    rule.format = hyperlinkFormat;
+    rule.matchType = QRegularExpression::NormalMatch;
+    highlightingRules.append(rule);
+
+    const int hyperlinksRuleIndex = highlightingRules.size() - 1;
+
+    //HIGHLIGHTED 
+    highlightedHyperlinkFormat.setForeground(m_hyperlinkColor);
+    highlightedHyperlinkFormat.setBackground(m_hyperlinkHoverColor);
+    rule.pattern = highlightedHyperlinkRegularExpression();
+    rule.format = highlightedHyperlinkFormat;
+    rule.matchType = QRegularExpression::NormalMatch;
+    highlightingRules.append(rule);
+
+    const int highlightedHyperlinkRuleIndex = highlightingRules.size() - 1;
+
+    connect(this, &StatusSyntaxHighlighter::hyperlinksChanged, this, [hyperlinksRuleIndex, this](){
+        highlightingRules[hyperlinksRuleIndex].pattern = hyperlinksRegularExpression();
+        rehighlight();
+    });
+    connect(this, &StatusSyntaxHighlighter::hyperlinkColorChanged, this, [hyperlinksRuleIndex, this](){
+        hyperlinkFormat.setForeground(m_hyperlinkColor);
+        highlightedHyperlinkFormat.setForeground(m_hyperlinkColor);
+        highlightingRules[hyperlinksRuleIndex].format = hyperlinkFormat;
+        rehighlight();
+    });
+
+    connect(this, &StatusSyntaxHighlighter::highlightedHyperlinkChanged, this, [highlightedHyperlinkRuleIndex, this](){
+        highlightingRules[highlightedHyperlinkRuleIndex].pattern = highlightedHyperlinkRegularExpression();
+        rehighlight();
+    });
+    connect(this, &StatusSyntaxHighlighter::hyperlinkHoverColorChanged, this, [highlightedHyperlinkRuleIndex, this](){
+        highlightedHyperlinkFormat.setBackground(m_hyperlinkHoverColor);
+        highlightingRules[highlightedHyperlinkRuleIndex].format = highlightedHyperlinkFormat;
+        rehighlight();
+    });
 }
 
 void StatusSyntaxHighlighter::highlightBlock(const QString& text)
 {
     for(const HighlightingRule& rule : qAsConst(highlightingRules))
     {
+        if(rule.pattern.pattern() == QStringLiteral("")) continue;
```

Comment:
```suggestion
        if(rule.pattern.pattern().isEmpty()) continue;
```

Reply:
Not sure it helps. Or I'm missing something. 🤔 

Looks to me like QUrl::fromUserInput will try to alter the url and we don't want that. It should be processed as received from the model so that it can match the input string.
I don't really want to validate it or to add anything extra to the url. I've used the QUrl just for the ease of processing the url with different encodings so that we don't assume anything about the backend encoding.

From their docs:
> Examples:
qt-project.org becomes http://qt-project.org
ftp.qt-project.org becomes ftp://ftp.qt-project.org
hostname becomes http://hostname
/home/user/test.html becomes file:///home/user/test.html
This function was introduced in Qt 4.6.

Reply:
I see, was just a suggestion 😊